metrics/web/docs/features/navigation/diagrams/navigation_class_diagram.puml

```diff
@@ -0,0 +1,114 @@
+@startuml navigation_class_diagram
+
+class MetricsApp {}
+
+package common.presentation {
+
+    package injector.widget {
+        class InjectionContainer {}
+    }
+
+    package navigation {
+
+        package route_configration {
+            class RouteName {
+            }
+
+            class RouteConfiguration {
+                + name : RouteName
+                + path : String
+                + authorizationRequired : bool
+            }
+
+            class RouteConfigurationFactory {
+                + RouteConfiguration create()
+            }
+        }
+        class MetricsRouteInformationParser {
+            - RouteConfigurationFactory _routeConfigurationFactory
+            + Future<RouteConfiguration> parseRouteInformation()
+            + RouteInformation restoreRouteInformation()
+        }
+        class MetricsRouterDelegate {
+            + navigatorKey : GlobalKey<NavigatorState>
+            + currentConfiguration : RouteConfiguration
+            + navigatorObservers : List<NavigatorObservers>
+            - navigationNotifier : NavigationNotifier
+            + Future<void> setInitialRoutePath(RouteConfiguration routeConfiguration)
+            + Future<void> setNewRoutePath(RouteConfiguration routeConfiguration)
+        }
+
+        package constants {
+            class MetricsRoutes {}
+        }
+
+        package state {
+            class NavigationNotifier {
+                + pages : List<MetricsPage>
+                + currentConfiguration : RouteConfiguration
+                - _metricsPageFactory : MetricsPageFactory
+                - _routeConfigurationFactory : RouteConfigurationFactory
+                - _isUserLoggedIn : bool
+                + void handleUserStateUpdates(bool isLoggedIn)
+                + void pop(MetricsPage page)
+                + void push(MetricsPage page)
+                + void pushNamed(String path)
+                + void pushReplacement(MetricsPage page)
+                + void pushReplacementNamed(String path)
+                + void pushAndRemoveUntil (MetricsPage page, bool Function(Page) test)
+                + void pushNamedAndRemoveUntil (MetricsPage page, bool Function(Page) test)
+                + void didPop(MetricsPage page)
+                + void setInitialRoutePath(RouteConfiguration routeConfiguration)
+                + void setNewRoutePath(RouteConfiguration routeConfiguration)
+            }
+        }
+
+
+        package metrics_page {
+            class MetricsPageFactory {
+                + MetricsPage<T> create<T>()
```

Comment:
```suggestion
                + MetricsPage<T> create<T>(RouteConfiguration)
```
As we don't use parameters with navigation - maybe this class shouldn't be generic then? 
```suggestion
                + MetricsPage create(RouteConfiguration)
```